### PR TITLE
Extracting Level Parsing from Level Class; Refactoring

### PR DESCRIPTION
**Pull request by W-McDonald**
The extraction seems to be working great, but we're still breaking tests. Will probably have to refactor and update the tests.

### CODE REVIEW

File: levels/level.py [hunk 1]
```diff
@@ -1,21 +1,14 @@
...
-        self._events: Dict = {}
-        self._level_name: str = ''
-        self._num_floors: int = 0
-        self._capacity: int = 0
-        self._level_finish_move_number: int = 0
-        self._parser_last_move_seen: int = -1
-
-        if filename != '':
-            self.parse_from_file(filename)
+    def __init__(self, events, level, num_floors, capacity, length_in_moves) -> None:
```
Comment (thbenschop) [inline levels/level.py]: types on args

File: levels/level_parser.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+class LevelParser(object):
+
+    def __init__(self, filename: str='') -> None:
+        self._filename: str = filename
+        self._parser_last_move_seen: int = -1
+
+    # TODO Add more features
+    def parse_from_file(self) -> Level:
+        # TODO Add return type to this function
```
Comment (thbenschop) [inline levels/level_parser.py]: remove todo

File: levels/level_parser.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+from levels.level import Level
+
+
+class LevelParser(object):
+
+    def __init__(self, filename: str='') -> None:
+        self._filename: str = filename
+        self._parser_last_move_seen: int = -1
+
+    # TODO Add more features
```
Comment (thbenschop) [inline levels/level_parser.py]: can probably go, this could go in every class

File: main.py [hunk 4]
```diff
@@ -5,14 +5,21 @@
...
 from controller.debug_controller import DebugGameController
 from levels.level import Level
+from levels.level_parser import LevelParser
 from model.model import BellhopModel
 from view.console_view import ConsoleView, PyGameViewTextWriter
 from view.debug_view import DebugView
 from view.view import BellhopView, PyGameImageWriter
 
 
+def get_parsed_level(level_path):
```
Comment (thbenschop) [inline main.py]: arg type

**Review by screwed99 — APPROVED** [review]
Looks great!

**Pull request merged**